main.cpp

```diff
@@ -0,0 +1,184 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <algorithm>
+
+using namespace std;
+
+class Animal
+{
+    public:
+        string Nume;
+        string Specie;
+        string Rasa;
+        string CategorieDeProductie;
+        int Greutate;
+        int Varsta;
+
+    Animal(){};
+    Animal(string nume, string specie, string rasa, string categorieDeProductie, int greutate, int varsta)
+        : Nume(nume), Specie(specie), Rasa(rasa), CategorieDeProductie(categorieDeProductie), Greutate(greutate), Varsta(varsta) {}
+
+};
+
+
+int showMenu()
+{
+    int number;
+    cout << "Alegeti una dintre optiuni:" << endl;
+    cout << "\t 1) Adaugati animal" << endl;
+    cout << "\t 2) Afisati lista animalelor" << endl;
+    cout << "\t 3) Editati detaliile unui animal" << endl;
+    cout << "\t 4) Eliminati un animal" << endl;
+    cout << "\t 5) Iesiti din aplicatie" << endl;
+    cin >> number;
+    return number;
+}
+
+// Function to convert a string to lowercase
+string toLowerCase(const string& str)
+{
+    string lowerStr = str;
+    transform(lowerStr.begin(), lowerStr.end(), lowerStr.begin(), ::tolower);
+    return lowerStr;
+}
```

Comment:
Nu stiu cat de mult sens are functia asta, ca practic nu ai economisit mare lucru. Este o linie de cod sa apelezi functia asta, si tot o linie de cod ar fi sa chemi direct `transform()` pe string-ul tau.
Asa mai se pierde timp sa se apeleze functia, sa se faca o copie a string-ului original, etc etc

Reply:
Nu imi dau seama cum sa modific aici fara sa lungesc codul. Functia asta e chemata de 2 ori in functia de mai jos, care verifica ambele string uri (cel din vector si cel bagat de utilizator) daca sunt la fel dupa ce au fost transformate. Si face copie daca parametrii sunt passed by reference?

Reply:
sincer nu-s sigur, dar tu pasezi `str` by reference, nu by value, asa e, dar dupa faci `lowerStr = str` si nu-s sigur daca asta face o copie sau nu.